tests/dispatch_group.c

```diff
@@ -19,7 +19,9 @@
  */
 
 #include <dispatch/dispatch.h>
+#if HAVE_UNISTD_H
 #include <unistd.h>
```

Comment:
for all of these because <dispatch/dispatch.h> will do the right thing we should just remove the `#include` instead

Reply:
typically nowadays (not merged back to github yet) Darwin does this to headers:

`unifdef -U__DISPATCH_BUILDING_DISPATCH__ -U__linux__ -DTARGET_OS_WIN32=0 -U__ANDROID__`

it allows to cleanup the headers from stuff that makes no sense on darwin to clean them up, I think we should do the same for various platforms to that headers for said plaforms are decluttered from stuff that makes no sense on said platform.

but we can't have '#if HAVE_*' on public/installed headers as we don't want to ship a `<dispatch/config.h>` (Darwin certainly doesn't so that change as is would break darwin).

Reply:
Hmm, another option is to use the compiler instead.  `unistd.h` is only available on Unix and macOS X.  We could also use:

    #if defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))

I need that adding the hooks for the `unifdef` should be done anyways as @das mentioned there was interest in unifying the macOS build into the CMake system as well.